pkg/framer/property_value_batch.go

```diff
@@ -53,4 +64,72 @@ func (p AssetPropertyValueBatch) Frames(ctx context.Context, resources resource.
 	}
 
 	return frames, nil
-}
\ No newline at end of file
+}
+
+func (AssetPropertyValueBatch) framePropertyValue(property *iotsitewise.DescribeAssetPropertyOutput, assetPropertyValue *iotsitewise.AssetPropertyValue) *data.Frame {
+	timeField := fields.TimeField(0)
+	valueField := fields.PropertyValueField(property, 0)
+	qualityField := fields.QualityField(0)
+
+	frame := data.NewFrame(*property.AssetName, timeField, valueField, qualityField)
+
+	if assetPropertyValue != nil {
+		timeField.Append(getTime(assetPropertyValue.Timestamp))
+		valueField.Append(getPropertyVariantValue(assetPropertyValue.Value))
+		qualityField.Append(*assetPropertyValue.Quality)
+	}
+	return frame
+}
+
+func (p AssetPropertyValueBatch) frameL4ePropertyValue(ctx context.Context, property *iotsitewise.DescribeAssetPropertyOutput, assetPropertyValue *iotsitewise.AssetPropertyValue, assetId *string) (*data.Frame, error) {
+	dataFields := []*data.Field{}
+
+	timeField := fields.TimeField(0)
+	dataFields = append(dataFields, timeField)
+
+	qualityField := fields.QualityField(0)
+	dataFields = append(dataFields, qualityField)
+
+	anomalyScoreField := fields.AnomalyScoreField(0)
+	dataFields = append(dataFields, anomalyScoreField)
+
+	predictionReasonField := fields.PredictionReasonField(0)
+	dataFields = append(dataFields, predictionReasonField)
+
+	if assetPropertyValue == nil {
+		frame := data.NewFrame(*property.AssetName, dataFields...)
+		return frame, nil
+	}
+
+	var l4eAnomalyResult models.L4eAnomalyResult
+	err := json.Unmarshal([]byte(*assetPropertyValue.Value.StringValue), &l4eAnomalyResult)
+	if err != nil {
+		return nil, err
+	}
+
+	timeField.Append(getTime(assetPropertyValue.Timestamp))
+	qualityField.Append(*assetPropertyValue.Quality)
+	anomalyScoreField.Append(l4eAnomalyResult.AnomalyScore)
+	predictionReasonField.Append(l4eAnomalyResult.PredictionReason)
+
+	for _, diagnostics := range l4eAnomalyResult.Diagnostics {
+		propertyId := strings.Split(diagnostics.Name, "\\")[0]
+
+		req := &iotsitewise.DescribeAssetPropertyInput{
+			AssetId:    assetId,
+			PropertyId: aws.String(propertyId),
+		}
+		resp, err := p.SitewiseClient.DescribeAssetPropertyWithContext(ctx, req)
```

Comment:
Yes, it would contribute to additional latency but no "cost" in term of billing. Unfortunately, this is the only API that provides the property name 🙏